server/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java

```diff
@@ -337,7 +369,14 @@ private HitContext prepareNestedHitContext(SearchContext context,
                 getInternalNestedIdentity(context, nestedDocId, subReaderContext, context.mapperService(), nestedObjectMapper);
 
         SearchHit hit = new SearchHit(nestedTopDocId, rootId, nestedIdentity, docFields, metaFields);
-        HitContext hitContext = new HitContext(hit, subReaderContext, nestedDocId, context.searcher(), sharedCache);
+        HitContext hitContext = new HitContext(
+            hit,
+            subReaderContext,
+            nestedDocId,
+            context.searcher(),
+            new SourceLookup(),  // Use a clean, fresh SourceLookup for the nested context
```

Comment:
why does it need to be clean rather than the one coming from the query shard context?

Reply:
I've not dug too deeply into it, but I *thought* we'd be trashing the data that we'd already loaded. I'm not 100% sure of that. I'll dig more.